plugin/src/main/java/org/opensearch/ml/task/MLTaskDispatcher.java

```diff
@@ -50,7 +51,7 @@ public MLTaskDispatcher(ClusterService clusterService, Client client, Settings s
         this.clusterService = clusterService;
         this.client = client;
         this.nodeHelper = nodeHelper;
-        this.maxMLBatchTaskPerNode = MLTaskManager.MAX_ML_TASK_PER_NODE;
+        this.maxMLBatchTaskPerNode = ML_COMMONS_MAX_ML_TASK_PER_NODE.get(settings);
         this.nextNode = new AtomicInteger(0);
         this.dispatchPolicy = ML_COMMONS_TASK_DISPATCH_POLICY.get(settings);
         clusterService.getClusterSettings().addSettingsUpdateConsumer(ML_COMMONS_TASK_DISPATCH_POLICY, it -> dispatchPolicy = it);
```

Comment:
Add similar update consumer for `ML_COMMONS_MAX_ML_TASK_PER_NODE` ? The consumer will listen to the update of `ML_COMMONS_MAX_ML_TASK_PER_NODE` and update value of maxMLBatchTaskPerNode.